sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java

```diff
@@ -767,6 +767,9 @@ public boolean start() throws IOException {
           reader.start()
               && rangeTracker.tryReturnRecordAt(true, ByteKey.of(reader.getCurrentRow().getKey()));
       if (hasRecord) {
+        if (recordsReturned == 0) {
+          source = source.withStartKey(rangeTracker.getStartPosition());
+        }
```

Comment:
I don't think this code needs to change. The range tracker could handle all of the range adjustment. Can you see why? (Maybe there's an issue somewhere else in BigtableIO?)